packages/admin/src/api/index.ts

```diff
@@ -0,0 +1,516 @@
+import type { ContextInterface } from '@textile/context'
+import {
+  privateKeyBytesToString,
+  publicKeyBytesToString,
+} from '@textile/crypto'
+import { GrpcConnection } from '@textile/grpc-connection'
+import * as pb from '@textile/hub-grpc/api/hubd/pb/hubd_pb'
+import { APIService } from '@textile/hub-grpc/api/hubd/pb/hubd_pb_service'
+import log from 'loglevel'
+
+const logger = log.getLogger('admin-api')
+
+export interface SigninOrSignupResponse {
+  key: string
+  session: string
+}
+
+/**
+ * Creates a new user (if username is available) and returns a session.
+ * @param username The desired username.
+ * @param email The user's email address.
+ * @param ctx Context containing gRPC headers and settings.
+ * These will be merged with any internal ctx.
+ * @note This method will block and wait for email-based verification.
+ * @internal
+ */
+export async function signUp(
+  api: GrpcConnection,
+  username: string,
+  email: string,
+  ctx?: ContextInterface,
+): Promise<SigninOrSignupResponse> {
+  logger.debug('signup request')
+  const req = new pb.SignupRequest()
+  req.setEmail(email)
+  req.setUsername(username)
+  const res: pb.SignupResponse = await api.unary(APIService.Signup, req, ctx)
+  return {
+    key: publicKeyBytesToString(res.getKey_asU8()),
+    session: res.getSession(),
+  }
+}
+
+/**
+ * Returns a session for an existing username or email.
+ * @param usernameOrEmail An existing username or email address.
+ * @param ctx Context containing gRPC headers and settings.
+ * These will be merged with any internal ctx.
+ * @note This method will block and wait for email-based verification.
+ * @internal
+ */
+export async function signIn(
+  api: GrpcConnection,
+  usernameOrEmail: string,
+  ctx?: ContextInterface,
+): Promise<SigninOrSignupResponse> {
+  logger.debug('signin request')
+  const req = new pb.SigninRequest()
+  req.setUsernameOrEmail(usernameOrEmail)
+  const res: pb.SigninResponse = await api.unary(APIService.Signin, req, ctx)
+  return {
+    key: publicKeyBytesToString(res.getKey_asU8()),
+    session: res.getSession(),
+  }
+}
+
+/**
+ * Deletes the current session.
+ * @param ctx Context containing gRPC headers and settings.
+ * These will be merged with any internal ctx.
+ * @internal
+ */
+export async function signOut(
+  api: GrpcConnection,
+  ctx?: ContextInterface,
+): Promise<void> {
+  logger.debug('signout request')
+  const req = new pb.SignoutRequest()
+  await api.unary(APIService.Signout, req, ctx)
+  return
+}
+
+export interface SessionInfoResponse {
```

Comment:
Each return type here is grouped and placed above the function that returns it..